src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -178,15 +178,34 @@ public CheckResult SetBindingInfo(RecordType parameterType)
             return this.SetBindingInfo(symbolTable);
         }
 
-        private FormulaType _expectedReturnType;
-        private bool _allowCoerceToType = false;
+        private FormulaType[] _expectedReturnTypes;
 
+        // This function only injects errors but does not do any coercion.
         public CheckResult SetExpectedReturnValue(FormulaType type, bool allowCoerceTo = false)
+        {
+            if (allowCoerceTo)
+            {
+                switch (type)
+                {
+                    case StringType:
+                        return SetExpectedReturnValue(StringValue.AllowedListConvertToString.ToArray());
+                    case NumberType:
+                        return SetExpectedReturnValue(NumberValue.AllowedListConvertToNumber.ToArray());

```

Comment:
>AllowedListConvertToNumber [](http://example.com/codeflow?start=66&length=26)

I know this is not part of those changes, but can you also update this to include FormulaType.Time (which can be converted to number)? Same with DecimalValue.AllowedListConvertToDecimal.

Reply:
Added datetime and boolean here.